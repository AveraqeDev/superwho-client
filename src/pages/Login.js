import React, { useState, useEffect } from 'react';

import { Input, Button } from '../components/Utils';

import '../styles/Login.css';
import { Link } from 'react-router-dom';

const SidePanel = () => {
  return (
    <div className='SidePanel'>
      <h1>SuperWho?</h1>
      <div className="SidePanel__message">
        <h2>Welcome back!</h2>
        <p>We are glad to see you again!  Please login using your username and password!</p>
      </div>
    </div>
  );
};

const LoginForm = () => {
  const [error, setError] = useState(null);
  
  const onSubmit = e => {
    e.preventDefault();
    setError(null);
    const { username, password } = e.target;

    console.log(username, password);
  };
  
  return (
    <form className='LoginForm' onSubmit={onSubmit}>
      <h2 className="Header">Log In</h2>
      <div role='alert'>
        {error && <p className='error'>{error}</p>}
      </div>

      <div className='username'>
        <label htmlFor='LoginForm__username'>
          Username
        </label>
        <Input 
          required
          name='username'
          id='LoginForm__username'
          placeholder='Enter Your Username'
        />
      </div>
      <div className='password'>
        <label htmlFor='LoginForm__password'>
          Password
        </label>
        <Input 
          required
          name='password'
          type='password'
          id='LoginForm__password'
          placeholder='Enter Your Password'
        />
      </div>

      <Button className='LoginForm__button' type='submit'>
        Login
      </Button>
      <p className='LoginForm__signup'>Don't have an account? <Link className='LoginForm__signup-link' to='/register'>Sign Up</Link></p>
    </form>
  );
};

const Login = () => {
  const [windowWidth, setWindowWidth] = useState(0);
  const resizeWindow = () => {
    setWindowWidth(window.innerWidth);
  };

  useEffect(() => {
    resizeWindow();
    window.addEventListener("resize", resizeWindow);
    return () => window.removeEventListener("resize", resizeWindow);
  }, []);

  if(windowWidth <= 760) {
    return (
      <div className='Login'>
        <LoginForm />
      </div>
    );
  } else {
    return (
      <div className='Login'>
        <SidePanel />
        <LoginForm />
      </div>
    );
  }
};

export default Login;