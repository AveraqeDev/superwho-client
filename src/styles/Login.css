.Login {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 100vw;
  height: 100vh;
}

.SidePanel {
  width: 50%;
  height: 100%;
  background: #2A9BE4;
  box-shadow: 4px 0 4px rgba(0, 0, 0, 0.25);
  color: #FFFFFF;
}

.SidePanel > h1 {
  font-size: 50px;
  padding: 2rem;
}

.SidePanel__message {
  padding: 5rem;
}

.SidePanel__message > h2 {
  margin-bottom: 0;
  font-size: 40px;
}

.SidePanel__message > p {
  font-size: 20px;
}

.LoginForm {
  width: 50%;
  padding: 0 5rem;
  margin: auto auto;
  min-width: 415px;
}

.Header {
  font-size: 40px;
  margin-bottom: 40px;
}

.username {
  width: 100%;
}

.username > label {
  font-size: 20px;
}

#LoginForm__username {
  width: 100%;
  margin: .5rem 0;
  padding: 12px;
  color: black;
}

.password > label {
  font-size: 20px;
}

#LoginForm__password {
  width: 100%;
  margin: .5rem 0;
  padding: 12px;
  color: black;
}

.LoginForm__button {
  width: 100%;
  font-size: 20px;
  padding: 7px;
  margin-top: 2rem;
}

.LoginForm__button:hover {
  opacity: 0.8;
}

.LoginForm__signup {
  text-align: center;
  color: #CBD0D6;
  font-weight: bold;
  font-size: 16px;
  padding: 1rem;
}

.LoginForm__signup-link {
  text-decoration: none;
  color: #2A9BE4;
}

.LoginForm__signup-link:hover {
  opacity: 0.8;
}

@media screen and (max-width: 900px) {
  .SidePanel__message {
    padding: 5rem 2.5rem;
  }
}

@media screen and (max-width: 760px) {
  .LoginForm {
    width: 100%;
    padding: 0 5rem;
  }
}