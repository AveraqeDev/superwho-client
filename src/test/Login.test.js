import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Login, { SidePanel, LoginForm } from '../pages/Login';

describe('Login', () => {
  describe('<SidePanel />', () => {
    // Smoke test
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<SidePanel />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    //Snapshot test
    it('renders the UI as expected', () => {
      const tree = renderer
        .create(
          <SidePanel />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    })
  });

  describe('<LoginForm />', () => {
    const testData = {
      onLogin: () => {}
    };
    // Smoke test
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <MemoryRouter>
          <LoginForm onLogin={testData.onLogin} />
        </MemoryRouter>, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    //Snapshot test
    it('renders the UI as expected', () => {
      const tree = renderer
        .create(
          <MemoryRouter>
            <LoginForm onLogin={testData.onLogin} />
          </MemoryRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    })
  });

  describe('<Login />', () => {
    // Smoke test
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <MemoryRouter>
          <Login />
        </MemoryRouter>, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    //Snapshot test
    it('renders the UI as expected', () => {
      const tree = renderer
        .create(
          <MemoryRouter>
            <Login />
          </MemoryRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    })
  });
});