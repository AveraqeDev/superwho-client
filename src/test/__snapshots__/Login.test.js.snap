// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login <Login /> renders the UI as expected 1`] = `
<div
  className="Login"
>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <h2
      className="Header"
    >
      Log In
    </h2>
    <div
      role="alert"
    />
    <div
      className="LoginForm__username-container"
    >
      <label
        htmlFor="LoginForm__username"
      >
        Username
      </label>
      <input
        className="Input "
        id="LoginForm__username"
        name="username"
        placeholder="Enter Your Username"
        required={true}
      />
    </div>
    <div
      className="LoginForm__password-container"
    >
      <label
        htmlFor="LoginForm__password"
      >
        Password
      </label>
      <input
        className="Input "
        id="LoginForm__password"
        name="password"
        placeholder="Enter Your Password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="Button LoginForm__button"
      type="submit"
    >
      Login
    </button>
    <p
      className="LoginForm__signup"
    >
      Don't have an account? 
      <a
        className="LoginForm__signup-link"
        href="/register"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </p>
  </form>
</div>
`;

exports[`Login <LoginForm /> renders the UI as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <h2
    className="Header"
  >
    Log In
  </h2>
  <div
    role="alert"
  />
  <div
    className="LoginForm__username-container"
  >
    <label
      htmlFor="LoginForm__username"
    >
      Username
    </label>
    <input
      className="Input "
      id="LoginForm__username"
      name="username"
      placeholder="Enter Your Username"
      required={true}
    />
  </div>
  <div
    className="LoginForm__password-container"
  >
    <label
      htmlFor="LoginForm__password"
    >
      Password
    </label>
    <input
      className="Input "
      id="LoginForm__password"
      name="password"
      placeholder="Enter Your Password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="Button LoginForm__button"
    type="submit"
  >
    Login
  </button>
  <p
    className="LoginForm__signup"
  >
    Don't have an account? 
    <a
      className="LoginForm__signup-link"
      href="/register"
      onClick={[Function]}
    >
      Sign Up
    </a>
  </p>
</form>
`;

exports[`Login <SidePanel /> renders the UI as expected 1`] = `
<div
  className="SidePanel"
>
  <h1>
    SuperWho?
  </h1>
  <div
    className="SidePanel__message"
  >
    <h2>
      Welcome back!
    </h2>
    <p>
      We are glad to see you again!  Please login using your username and password!
    </p>
  </div>
</div>
`;
